### PR TITLE
Fix media type for queryables endpoints

### LINKED ISSUE #400: Content type of /queryables endpoint is application/json instead of application/schema+json

### PR DESCRIPTION
**Pull request by duckontheweb**
**Related Issue(s):** 

- Closes #400 

**Description:**

Creates a new `JSONSchemaResponse` class with a `application/schema+json` media type that is used in the `/queryables` responses.

**PR Checklist:**

- [x] Code is formatted and linted (run `pre-commit run --all-files`)
- [x] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/master/CHANGES.md).

### CODE REVIEW

**Review by geospatial-jeff — APPROVED** [review]
LGTM!

**Comment by duckontheweb** [comment]
@geospatial-jeff What's the convention for merging these PRs. Is there a maintainer who should do it, or is the author expected to merge them once they are approved?

**Comment by geospatial-jeff** [comment]
> What's the convention for merging these PRs.

Honestly not sure, we should probably figure this out haha.  Created https://github.com/stac-utils/stac-fastapi/issues/433

**Review by geospatial-jeff — APPROVED** [review]

**Pull request merged**
